### ISSUE TITLE
Add suport for externally linked contact photos

### ISSUE THREAD

**Issue by laspic** [jobisoft/TbSync#351]
I hope that the question has not yet been asked (but I found nothing in the issues or the wiki).

I use Thunderbird (68.2.2) and TBsync (2.9) to synchronize contacts from a personal Nextcloud (17.0.2) server and everything works well. 

I'm surprised that the contact's photos I add locally are not synchronized with TBsync. Every field is send to Nextcloud, except the photo.

Is it an issue ? Or the feature for later ?

**Comment by jobisoft** [comment]
Looks like a bug. I will check.

**Comment by jobisoft** [comment]
Can I have a test account on your NC instance? My Testserver is down so I cannot analyse this issue without access to a affected server.

**Comment by jobisoft** [comment]
My testserver is up again and I cannot confirm you findings. I added a photo to one of my contacts and it is synced to NC. 

My testserver is NC 17.0.3. What is yours?

**Comment by laspic** [comment]
Sure, send me your email address and I create you an test account. 

The version is 17.0.1, I'm launching now an update to 17.0.3.

**Comment by jobisoft** [comment]
Ok, my email is listed here:
https://github.com/jobisoft/

**Comment by laspic** [comment]
Done, you should have received the welcome message.
My server has been updated to 17.0.3.

**Comment by jobisoft** [comment]
Works for me:

![grafik](https://user-images.githubusercontent.com/5830621/73594831-46eebb80-4512-11ea-8294-fcdfd55ceed9.png)

1. Created the contact in TB without photo
2. synced and checked on server
3. added photo in TB
4. synced and checked on server

Can you verify?

**Comment by jobisoft** [comment]
Can you confirm, that it works with your server? What is your status? It worked for me and i tend to close this soon.

**Comment by GUAYMI** [comment]
Yes

Obtener Outlook para Android<https://aka.ms/ghei36>
________________________________

**Comment by jobisoft** [comment]
@

**Comment by laspic** [comment]
Sorry for the delay in responding, I was a bit busy at work.

Thank you for your tests, which led me to investigate deeper. You are right, in the most cases, it works well. As a matter of fact, the issue appears only when I add a photo not using the field "On this computer" **but the field "On the web"**. 

- if a new contact has just been created, and I change the generic photo by another one located on the web → the new one is ignored and the generic one remains the one synchronized ;
- if the contact has already had a personalized photo, and I change it by one located on the web → the new one is ignored and the previous one remains the one synchronized.

In conclusion, the synchronization fails in the case when I add to a contact the photo he or she has on their Twitter profile directly by the photo's weblink.

Can you reproduce that bug on your side ?

**Comment by jobisoft** [comment]
@laspic : Could you create a new issue specific for the new finding/bug?

**Comment by jobisoft** [comment]
@laspic : No need to open a new issue. I understood now what this is about and I am close to have support for externally linked photos done. Will report here.

**Comment by jobisoft** [comment]
I think it is working in the latest beta:
https://tbsync.jobisoft.de

Please give it a try! If you are on beta already, just enforce an update of your addons. You probably should disable and re-enable the TbSync account once, to activate the new feature.

I saw, that NC does not display externaly linked photos in its own web UI, but other clients see them.

**Comment by laspic** [comment]
Thank you very much for the enhancement! :+1: 
I'm really sorry however but I have no time now to test it (which would imply to create a test profile, set up a minimum working configuration and install the beta). I have to trust you. I'm waiting so for the next stable version!

**Comment by jobisoft** [comment]
You would not need to setup email, just the profile and the tbsync accont. 10min at most. Having this confirmed before it goes life would be very helpful.

**Comment by laspic** [comment]
Indeed, a light configuration was sufficient.

I don't know if there is a link with the changes you made but a problem occurs with the synchronization: only some contacts are downloaded (37/109). It tells me that it encounters a "javascript error". Here is the log details:

> vCardData.photo[0].value.split is not a function
setThunderbirdCardFromVCard@chrome://dav4tbsync/content/includes/tools.js:922:69
addContact@chrome://dav4tbsync/content/includes/tools.js:359:29
multiget@chrome://dav4tbsync/content/includes/sync.js:716:49

Does it make sense for you ?

**Comment by jobisoft** [comment]
I will have a look. Thanks for testing!

**Comment by jobisoft** [comment]
I would Need a bug Report to understand this. This should need not more than 5-10 minutes:

* enable debug logging in TbSync account manager -> help
* disable your account in TbSync
* restart Thunderbird
* enable your account and sync, to trigger the error again
* create a bug report in TbSync account manager -> help

Thanks for your time!

**Comment by laspic** [comment]
Done. But I had a look on the file .log, I'm not sure it gives you much more details.

**Comment by jobisoft** [comment]
It does not, because I forgot to tell you to activate debug logging in the highest level. :-( I replied to your bug report with instructions. Sorry for that.

**Comment by laspic** [comment]
Done.  

/!\ Thank you to delete this file after debugging, as soon as possible: as I can see it contains **a lot of** private data!

**Comment by jobisoft** [comment]
I think I fixed it:
1. disable your tbsync account
2. update your add-ons via the gear menu in the add-on manager
3. enable your tbsync account  and try again to sync

**Comment by laspic** [comment]
Everything's good now! 

- synchronization: OK (no bug anymore)
- linked photo feature: OK (indeed the photo is not displayed on my NC server but it is on my phone).

Many thanks !
